sys/include/net/ng_neterr.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_ng_neterr   Error reporting
+ * @ingroup     net
+ * @brief       Allows for asynchronous error reporting in the network stack.
+ * @{
+ *
+ * @file
+ * @brief   Error reporting definitions.
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef NG_NETERR_H_
+#define NG_NETERR_H_
+
+#include <errno.h>
+#include <stdint.h>
+
+#include "kernel_types.h"
+#include "msg.h"
+#include "net/ng_pkt.h"
+#include "thread.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   @ref core_msg type for reporting an error.
+ */
+#define NG_NETERR_MSG_TYPE          (0x0206)
+
+/**
+ * @brief   Error code for success.
+ */
+#define NG_NETERR_SUCCESS           (0)
+
+/**
+ * @brief   Reports an error to all subscribers of errors to @p pkt.
+ *
+ * @param[in] pkt   Packet snip to report on.
+ * @param[in] err   The error code for the packet.
+ */
+#ifdef MODULE_NG_NETERR
+static inline void ng_neterr_report(ng_pktsnip_t *pkt, uint32_t err)
+{
+    if (pkt->err_sub != KERNEL_PID_UNDEF) {
+        msg_t msg;
+
+        msg.type = NG_NETERR_MSG_TYPE;
+        msg.content.value = err;
+
+        msg_send(&msg, pkt->err_sub);
+    }
+}
+#else
+#define ng_neterr_report(pkt, err)  (void)pkt; (void)err;
```

Comment:
I got an error (or better a warning that became an error due to `-Werror`) with that.